src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -334,8 +350,35 @@ public int onErrorHandling() {
       return 1;
     }
 
+    /**
+     * Gets the custom template specified by user which can be either a templateId or a full custom template path
+     *
+     * @return String of templateId or custom template path
+     */
+    public String getCustomTemplate() {
+      return Strings.isNullOrEmpty(templateId) ? customTemplatePath :
+        String.format("projects/%s/inspectTemplates/%s", getProject(), templateId);
+    }
 
     public void validate(FailureCollector collector, Schema inputSchema) {
+      if (customTemplateEnabled) {
+        if (!containsMacro(CUSTOM_TEMPLATE_PATH_NAME) && Strings.isNullOrEmpty(customTemplatePath)) {
+          if (!containsMacro(TEMPLATE_ID_NAME) && Strings.isNullOrEmpty(templateId)) {
```

Comment:
Can you move the nested condition form nested `if` to first `if`?

Reply:
PR updated.

Reply:
PR updated.